File: database/src/main/java/org/difft/app/database/models/ReadInfoModel.java
package org.difft.app.database.models;

import com.tencent.wcdb.MultiPrimary;
import com.tencent.wcdb.WCDBField;
import com.tencent.wcdb.WCDBIndex;
import com.tencent.wcdb.WCDBTableCoding;

import java.util.Objects;

@WCDBTableCoding(multiPrimaries = @MultiPrimary(columns = {"roomId", "uid"}))
public class ReadInfoModel {

    @WCDBIndex
    @WCDBField(isNotNull = true)
    public String roomId;

    @WCDBField(isNotNull = true)
    public String uid;

    @WCDBField
    public long readPosition; //读的位置

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        ReadInfoModel that = (ReadInfoModel) o;
        return readPosition == that.readPosition && Objects.equals(roomId, that.roomId) && Objects.equals(uid, that.uid);
    }

    @Override
    public int hashCode() {
        return Objects.hash(roomId, uid, readPosition);
    }
}

File: database/src/main/java/org/difft/app/database/models/ReadInfoModel.kt
package org.difft.app.database.models

import com.tencent.wcdb.MultiPrimary
import com.tencent.wcdb.WCDBField
import com.tencent.wcdb.WCDBIndex
import com.tencent.wcdb.WCDBTableCoding
import java.util.Objects

@WCDBTableCoding(multiPrimaries = [MultiPrimary(columns = ["roomId", "uid"])])
class ReadInfoModel {

    @WCDBIndex
    @WCDBField(isNotNull = true)
    var roomId: String = ""

    @WCDBField(isNotNull = true)
    var uid: String = ""

    @WCDBField
    var readPosition: Long = 0 //读的位置

    override fun equals(other: Any?): Boolean {
        if (other == null || javaClass != other.javaClass) return false
        other as ReadInfoModel
        return readPosition == other.readPosition && roomId == other.roomId && uid == other.uid
    }

    override fun hashCode(): Int = Objects.hash(roomId, uid, readPosition)
}
